test/integration/targets/azure_rm_eventhub/tasks/main.yml

```diff
@@ -0,0 +1,264 @@
+  set_fact:
+      namespace_aname: "eventhubnamespace{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}a"
+      namespace_bname: "eventhubnamespace{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}b"
+      eventhub_aname: "eventhub{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}a"
+      eventhub_bname: "eventhub{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}b"
+  run_once: yes
+
+- name: Create an eventhub namespace with default settings
+  azure_rm_eventhubnamespace:
+    name: "{{ namespace_aname }}"
+    location: eastus
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert the eventhub namespace was created
+  assert:
+    that: output.changed
+
+- name: Create an eventhub namespace for listing facts
+  azure_rm_eventhubnamespace:
+    name: "{{ namespace_bname }}"
+    location: eastus
+    resource_group: "{{ resource_group }}"
+    is_auto_inflate_enabled: true
+    maximum_throughput_units: 8
+  register: output
+
+- name: Assert the eventhub namespace was created
+  assert:
+    that: output.changed
+
+- name: Update an event hub namespace
+  azure_rm_eventhubnamespace:
+    name: "{{ namespace_aname }}"
+    location: eastus
+    resource_group: "{{ resource_group }}"
+    is_auto_inflate_enabled: true
+    maximum_throughput_units: 4
+  register: output
+
+- name: Assert the eventhub namespace was updated
+  assert:
+    that: 
+      - output.changed
+      - output.maximum_throughput_units == 4
+
+- name: Update an eventhub namespace (idempotent)
+  azure_rm_eventhubnamespace:
+    name: "{{ namespace_aname }}"
+    location: eastus
+    resource_group: "{{ resource_group }}"
+    is_auto_inflate_enabled: true
+    maximum_throughput_units: 4
+  register: output
+
+- name: Assert the eventhub namespace was updated
+  assert:
+    that: not output.changed
+
+- name: Get facts for an eventhub namespace
+  azure_rm_eventhubnamespace_facts:
+    name: "{{ namespace_aname }}"
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert that facts are returned
+  assert:
+    that:
+       - not output.changed
+       - output.eventhubnamespaces[0]['id'] != None
+       - output.eventhubnamespaces[0]['name'] != None
+       - output.eventhubnamespaces[0]['location'] != None
+       - output.eventhubnamespaces[0]['sku'] != None
+       - output.eventhubnamespaces[0]['is_auto_inflate_enabled'] != None
+       - output.eventhubnamespaces[0]['maximum_throughput_units'] != None
+       - output.eventhubnamespaces[0]['kafka_enabled'] != None
+
+- name: Get all facts for eventhub namespace in the resource group
+  azure_rm_eventhubnamespace_facts:
+    resource_group: "{{ resource_group }}"
+  register: output
+
+- name: Assert that facts are returned
+  assert:
+    that:
+       - not output.changed
+       - output.eventhubnamespaces[0]['id'] != None
```

Comment:
You can assert the length as `output.eventhubnamespaces | length > 1`

Reply:
Thank you, yuwei! I will improve the code based on your comments